Feature/54 beneficiary editing and deletion

JaviGarc1a opened the pull request:
# Description

Implemented the edit and delete beneficiary features. When clicking on the edit button you can see that the button changes, since with this button the form submission is confirmed, if nothing is filled in it remains as it was. When you click on the delete button you get a modal asking for confirmation and if yes, the beneficiary will be deleted. The rehabilitated feature can be modified in the edit view and in the normal view.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# Screenshot:
|Detail beneficiary|Edit beneficiary|
|-----------------------|---------------------|
|![image](https://github.com/ISPP-07/frontend-acat/assets/72880234/a1294195-f193-49fe-8fa2-5cb172d7031c)|![image](https://github.com/ISPP-07/frontend-acat/assets/72880234/85aeb54e-4a18-4268-9227-bb300e67aeea)|
Closes #54

JaviGarc1a committed:
fix: Backend change URL

JaviGarc1a committed:
styles: Conditional rendering to edit mode in patient

JaviGarc1a committed:
feat: Add prop isSubmit to ButtonIcon

JaviGarc1a committed:
feat: Add confirmation modal

JaviGarc1a committed:
feat: Edit and delete patient

JaviGarc1a committed:
fix: Responsive modal confirmation

JaviGarc1a committed:
fix: Add toggle delete in button icon

JaviGarc1a committed:
fix: Load page when edit

JaviGarc1a committed:
feat: Toggle button to rehabilitate patient

JaviGarc1a committed:
fix: Fix style toggle button

JaviGarc1a committed:
fix: Add validation DNI, NIE and passport

JaviGarc1a committed:
fix: form validation and error handling

JaviGarc1a committed:
feat: Update beneficiary birth date and first technician

JaviGarc1a committed:
fix: Sidebar show or hidden depend screen layout

JMGarCas reviewed (changes requested):
Good morning Javier. There a couple of things that have to be improved:
- [x] Although in the images you have shared you can see the icons in the buttons, In my screen those icon does not appear. I think this is due to styling issues.
- [x] The update form accepts blank spaces.
- [x] There is no date validation.
- [x] The page.jsx file is way too long. Try to create some components to reduce the length of the code.

JaviGarc1a committed:
fix: Add validation fields and correct style issue of button

JaviGarc1a committed:
refactor: Abstract Beneficiary details in 2 compontents

BeneficiaryDetails has been divided in two component BeneficiaryDetailsEdit and BeneficiaryDetailsView

JaviGarc1a committed:
Merge branch 'develop' into feature/54-beneficiary_editing_and_deletion

JMGarCas committed:
test: beneficiaryId (not finished)

JMGarCas committed:
Merge branch 'feature/54-beneficiary_editing_and_deletion' of https://github.com/ISPP-07/frontend-acat into feature/54-beneficiary_editing_and_deletion

JaviGarc1a committed:
fix: Change font size input name

JaviGarc1a committed:
fix: Delete @ imports

JMGarCas committed:
test: beneficiaryId (not finished yet)

JMGarCas committed:
Merge branch 'feature/54-beneficiary_editing_and_deletion' of https://github.com/ISPP-07/frontend-acat into feature/54-beneficiary_editing_and_deletion

JMGarCas committed:
test: beneficiaryId

JMGarCas committed:
fix: delete .yaml to fix deploy

JaviGarc1a committed:
fix: Change format date

JaviGarc1a committed:
Merge branch 'develop' into feature/54-beneficiary_editing_and_deletion

JMGarCas committed:
fix: tests

JMGarCas reviewed (approved):
Great job!
